### PR TITLE
[EngSys] removing steps to create go workspace

### PR DESCRIPTION
**Pull request by seankane-msft**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

Fixes https://github.com/Azure/azure-sdk-for-go/issues/16072

### CODE REVIEW

File: eng/pipelines/templates/steps/build-test.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ steps:
       pwsh: true
       filePath: eng/scripts/build.ps1
       arguments: -filter '${{ parameters.ServiceDirectory }}'
-      workingDirectory: '${{ parameters.GoWorkspace }}'
+      workingDirectory: $(System.DefaultWorkingDirectory)
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/build-test.yml:31]: You shouldn't need to explicitly set this as that is the default.

File: eng/pipelines/templates/steps/build-test.yml [hunk 2]
```diff
@@ -68,30 +68,30 @@ steps:
...
       env:
         GO111MODULE: 'on'
         PROXY_CERT: $(Build.SourcesDirectory)/eng/common/testproxy/dotnet-devcert.crt
         ${{ insert }}: ${{ parameters.EnvVars }}
 
     - pwsh: ../eng/scripts/create_coverage.ps1 ${{parameters.ServiceDirectory}}
       displayName: 'Generate Coverage XML'
-      workingDirectory: '${{parameters.GoWorkspace}}sdk'
+      workingDirectory: $(System.DefaultWorkingDirectory)/sdk
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/build-test.yml:79]: We should look at remove the "sdk" path here and let the create_coverage script understand that is needed with a servicedirectory.

**Comment by benbp** [comment]
@seankane-msft the publish test results search path needs to be updated too: https://github.com/Azure/azure-sdk-for-go/blob/main/eng/pipelines/templates/steps/build-test.yml#L79

**Review by benbp — APPROVED** [review]

**Pull request merged**